src/models/rule_based/model.py

```diff
@@ -0,0 +1,159 @@
+    def __init__(self,  path: str="."):
+        self._done = True
+        self._collected_data = self._init_data()
+        self._stage = 0
+        "possible stages:\n"
+        "        0. ask to confirm rule_based start\n"
+        "        1. confirm start\n"
+        "        2. ask for information\n"
+        "           provide answer\n"
+        "        3. confirm end\n"
+        "           ask for rechecking\n"
+        "        4. recheck information\n"
+        self._current_field = None
+        self.schedule = pd.merge(pd.merge(pd.read_csv(f"{path}/data/rule_based/schedule.csv", header=0),
+                                          pd.read_csv(f"{path}/data/rule_based/course.csv", header=0)),
+                                 pd.read_csv(f"{path}/data/rule_based/tutors.csv", header=0))
+
+    def is_completed(self):
+        return self._done
+
+    @staticmethod
+    def _init_data():
+        return dict(
+            course=dict(
+                data_type=int,
+                value=None,
+                question="O jaki przedmiot pytasz?",
+                check=False
+            ),
+            tutor=dict(
+                data_type=int,
+                value=None,
+                question="Kto prowadzi zajęcia?",
+                check=False
+            ),
+            day=dict(
+                data_type=int,
+                value=None,
+                question="Którego dnia odbywają się zającia?",
+                check=False
+            ),
+            start_hour=dict(
+                data_type=int,
+                value=None,
+                question="O której rozpoczynają się zającia?",
+                check=False
+            ),
+            group=dict(
+                data_type=int,
+                value=None,
+                question="O którą grupę pytasz?",
+                check=False
+            )
+        )
+
+    def _detect_confirmation(self, prompt: str) -> bool | type(None):
+        result = None
+        if 'tak' in prompt.lower():
+            result = True
+        elif 'nie' in prompt.lower():
+            result = False
+        return result
+
+    def _ask_for_data(self):
+        for k, v in self._collected_data.items():
+            if v["value"] is None:
+                self._current_field = k
+                return v['question']
+
+    def _retrive_info(self, prompt: str) -> type(None):
+        # needs to be a bit more readable but works
+        if self._current_field is not None:
+            for v in self.schedule[self._current_field].values:
+                # TODO look for aliases
+                aliases = [c for c in self.schedule.columns if re.compile(self._current_field).match(c)]
+                for v2 in self.schedule.loc[self.schedule[self._current_field] == v, aliases].values.reshape(-1):
+                    if str(v2).lower() in prompt.lower():
+                        self._collected_data[self._current_field]['value'] = v
+        return None
```

Comment:
I think you don't need to explicitely `return None`, if your method does not specify any return statement it it is going to do so automatically.
```suggestion
```